book/src/development/defining_lints.md

```diff
@@ -0,0 +1,164 @@
+# Define New Lints
+
+The first step in the journey of a new lint is the definition
+and registration of the lint in Clippy's codebase.
+We can use the Clippy dev tools to handle this step since setting up the 
+lint involves some boilerplate code.
+
+In our example, we're going to create a lint to detect functions named `foo` because it is a highly non-descriptive
+name for a function, so we want to trigger this and fix it early in the development process.
+
+## Lint name
+
+A good lint name is important, it is usually given by the issue you're fixing (in the **Lint name** field). If you're
+unsure if the name you chose fits the lint, you can check the [lint naming guidelines][lint_naming]. Don't worry, if
+the lint name doesn't fit, a Clippy team member will alert you in the PR process.
+
+If you're still unsure, you can ask on the [Zulip] or on the Github issue / PR.
+
+---
+
+We'll name our example lint that detects functions named "foo" `foo_functions`. Check the
+[lint naming guidelines][lint_naming] to see why this name makes sense.
+
+## Add and Register the Lint
+
+Now that a name is chosen, we shall register `foo_functions` as a lint to the codebase.
+There are two ways to register a lint.
+
+### Standalone
+
+If you believe that this new lint is a standalone lint (that doesn't belong to any specific [type](#lint-types) like `functions` or `loops`), you can run the following
+command in your Clippy project:
+
+```sh
+$ cargo dev new_lint --name=foo_functions --pass=late --category=pedantic
+```
+
+There are two things to note here:
+
+1. We set `--pass=late` in this command to do a late lint pass. The alternative
+is an `early` lint pass. We will discuss this difference in a later chapter.
+<!-- FIXME: Link that "later chapter" when lint_passes.md is merged -->
+2. If not provided, the `category` of this new lint will default to `nursery`.
+See Clippy's [lint types](../lints.md) for more information on categories.
```

Comment:
Here we have `lint types` linking to the upper chapter, and elsewhere there's a Lint Types heading, which makes me think the term might be overloaded. If so, could we find a more apt term here?

Reply:
Maybe changing that to link to the later section "Lint types", and adding a note in "Lint types" about how there's more information in `../lints.md` could clear the confusion?

Reply:
I think that the latest commit would solve the issue.